language/tools/cost_synthesis/src/global_state/account.rs

```diff
@@ -18,18 +22,21 @@ pub struct Account {
     /// The account address
     pub addr: AccountAddress,
     /// The account private key
-    pub privkey: PrivateKey,
+    pub privkey: Ed25519PrivateKey,
     /// The account public key
-    pub pubkey: PublicKey,
+    pub pubkey: Ed25519PublicKey,
     /// The set of modules that are published under that account.
     pub modules: Vec<VerifiedModule>,
 }
 
 impl Account {
     /// Create a new Account. The account is a logical entity at this point
     pub fn new() -> Self {
-        let (privkey, pubkey) = crypto::signing::generate_keypair();
-        let addr = pubkey.into();
+        let mut seed_rng = OsRng::new().expect("can't access OsRng");
```

Comment:
If we rely on randomness, here, would it make sense to have a central rng in the guise of `test_env.accounts.randomness_source`?

Reply:
As this is NOT for testing purposes, a central RNG would require a careful design, especially if we utilize multiple threads. I would prefer not changing it here and we can revisit our general RNG strategy in an upcoming PR. Wdyt?

Reply:
Fair enough! /cc @gedigi (who was looking into RNGs)